src/containers/Home.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+import { fetchPetitions, clearPetitions } from 'actions/PetitionsActions';
+import Home from 'components/Home';
+import getPetitions from 'selectors/petitions';
+import getPetitionsHomeQuery from 'selectors/petitionsHomeQuery';
+import isClientSideRouting from 'helpers/isClientSideRouting';
+
+const HOME_QUERY = getPetitionsHomeQuery({ limit: 3 });
+
+const HomeContainer = withRouter(React.createClass({
+
+  componentWillMount () {
+    // If there are no petitions, or if the user arrived on the page by clicking
+    // a client-side router link, then we fetch petitions client-side.
+    if (!this.props.petitions.length || isClientSideRouting(this.props.location)) {
+      const location = Object.assign({}, this.props.location, HOME_QUERY);
+
+      this.props.clearPetitions();
+      this.props.fetchPetitions({ location });
+    }
+  },
+
+  render () {
+    return (
+      <Home {...this.props} />
+    );
+  }
+}));
+
+HomeContainer.fetchData = (props) => {
+  const { store } = props;
+  const location = Object.assign({}, props.location, HOME_QUERY);
+
+  return store.dispatch(fetchPetitions({ location }));
```

Comment:
Still a bit unsure of this but not 100% on the best solution. I think we should make a new action called `fetchLatestPetitions(limit)` or something, which uses `fetchPetitions` inside that. Abstract away the location / logic from the component.